### PR TITLE
Update API for Grocery Lists

### PR DESCRIPTION
**Pull request by indirap**

### CODE REVIEW

**Comment by indirap** [comment]
TODO
- [x] Add tests

File: imports/api/groceries.js [hunk 1]
```diff
@@ -8,11 +8,17 @@ Meteor.publish('groceries', function() {
 	return Groceries.find({ owner: this.userId });
 });
 
+const authCheck = (groceryId) => {
```
Comment (vatthikorn) [inline imports/api/groceries.js]: This is neat. I probably should just pull this out to some sort of util class or somewhere that can be accessed for other APIs as well.

**Review by vatthikorn — APPROVED** [review]

File: imports/api/groceries.js (hunk 1 above)
Comment (indirap) [reply imports/api/groceries.js]: Yeah! I can put it in a separate file so we can reuse it.

File: utils/authorization.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Meteor } from 'meteor/meteor';
+
+const authCheck = (collection, userId, groceryId) => {
+	// Make sure only the owner can delete
+	const grocery = collection.findOne(groceryId);
+	if (grocery.owner !== userId) {
+		throw new Meteor.Error('not-authorized');
+	}
+}
```
Comment (vatthikorn) [inline utils/authorization.js]: These references to grocery should probably be changed to something more generic

**Review by vatthikorn — CHANGES_REQUESTED** [review]

**Review by vatthikorn — APPROVED** [review]

**Pull request merged**